### PR TITLE
py-rtree: fixes for standalone modules.

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Review by lguyot — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/py-rtree/package.py [hunk 1]
```diff
@@ -16,9 +16,16 @@ class PyRtree(PythonPackage):
...
+    def setup_build_environment(self, env):
         lib = self.spec['libspatialindex'].prefix.lib
-        spack_env.set('SPATIALINDEX_LIBRARY',
-                      join_path(lib, 'libspatialindex.%s'   % dso_suffix))
-        spack_env.set('SPATIALINDEX_C_LIBRARY',
-                      join_path(lib, 'libspatialindex_c.%s' % dso_suffix))
+        env.set('SPATIALINDEX_LIBRARY',
+                join_path(lib, 'libspatialindex.%s'   % dso_suffix))
+        env.set('SPATIALINDEX_C_LIBRARY',
+                join_path(lib, 'libspatialindex_c.%s' % dso_suffix))
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-rtree/package.py:24]: If these are standard variables and could be used by other packages. we could use `setup_dependent_build_environment` in the `libspatialindex` package itself. Another reason for that is when we make `libspatialindex` as external, we don't have `libspatialindex` in dependency spec. (hypothetical scenario)

**Review by pramodk — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-rtree/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-rtree/package.py:24]: Ah yes, I forgot that that would be the proper way.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-rtree/package.py:24]: Other references: https://github.com/Eyescale/CMake/blob/4a3414b5e95eba3bc9a56c668b8fc2fae0216e93/FindSpatialIndex.cmake#L23

I think it is OK to do it like this here. Other dependencies of `libspatialindex` should get similar settings passed at *build* time by Spack. This is specific to the Python package that we need them at *run* time.

**Review by jorblancoa — APPROVED** [review]

**Pull request merged**